data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractProcessor.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.model.processor;
+
+import com.amazon.dataprepper.metrics.MetricNames;
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.prepper.Prepper;
+import com.amazon.dataprepper.model.record.Record;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Timer;
+
+import java.util.Collection;
+
+/**
+ * Abstract implementation of the {@link Processor} interface. This class implements an execute function which records
+ * some basic metrics. Logic of the execute function is handled by extensions of this class in the doExecute function.
+ */
```

Comment:
Let's add the `@since` annotation for all the new java doc for both the AbstractProcessor and Processor class and methods.

Reply:
Good idea, `@since` annotation added.

Reply:
I only see the `@since` added to the AbstractProcessor. Can you add it to the `Processor` as well? Additionally, can you add the annotations to the methods? This will help differenciate support if these classes/interfaces are extended.

Reply:
Annotations added.